### PR TITLE
Add multi_assoc function

### PR DESCRIPTION
**Pull request by vichle**
This allows you to easily create arrays of associated objects.
 Example provided in README.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -40,6 +40,24 @@
       };
     };
 
+    factory.multi_assoc = function(numElements, name, key, attrs){
```
Comment (aexmachina) [inline index.js]: For consistency with the rest of the API this should be called `assocMany`.

File: index.js [hunk 2]
```diff
@@ -40,6 +40,24 @@
...
+      attrs = attrs || {};
+      return function(callback) {
+        results = [];
+        for(var i = 0; i < numElements; ++i)
+          results[i] = null;
```
Comment (aexmachina) [inline index.js]: Braces here please :)

File: index.js [hunk 3]
```diff
@@ -40,6 +40,24 @@
...
+        asyncForEach(results, function(elem, cb, index){
```
Comment (aexmachina) [inline index.js]: Space before brace please.

File: index.js [hunk 4]
```diff
@@ -40,6 +40,24 @@
...
+          factory.create(name, attrs, function(err, doc) {
+            if(err) return cb(err);
```
Comment (aexmachina) [inline index.js]: Space before brace here.

**Comment by vichle** [comment]
Thanks for the comments, I think I have addressed your concerns. I am quite new to JS and this is also my first contribution on GitHub so please bear with me :)

File: index.js [hunk 5]
```diff
@@ -40,6 +40,25 @@
       };
     };
 
+    factory.assocMany = function(numElements, name, key, attrs) {
```
Comment (aexmachina) [inline index.js]: This method should take args `function(name, key, num, attrsArray)` for consistency with the other methods. Sorry, I should have picked this up before.

**Comment by aexmachina** [comment]
Otherwise, LGTM.

**Comment by vichle** [comment]
I realized that I did not handle cases where the key argument was set. Also realised I re-invented the wheel and am now using the createMany function instead of creating many on my own :)

File: README.md [hunk 6]
```diff
@@ -42,10 +42,16 @@ factory.build('user', function(err, user) {
...
+factory.define('comment', Comment, {
+  text: 'Some constructive criticism'
+});
+
 factory.define('post', Post, {
   // create associations using factory.assoc(model, key)
   // or factory.assoc('user') to return the user object itself.
   user_id: factory.assoc('user', 'id'),
+  // create array of associations using factory.assocMany
+  comments: factory.assocMany(3, 'comment') // Will contains 3 elements
```
Comment (aexmachina) [inline README.md]: This usage example needs to be updated too.

File: index.js [hunk 7]
```diff
@@ -40,6 +40,27 @@
...
+    factory.assocMany = function(name, key, num, attrsArray) {
+      if (arguments.length < 4) {
+        if (typeof key === 'number') {
+          attrsArray = num;
+          num = key;
+          key = null;
+        }
+      }
+      return function(callback) {
+        factory.createMany(name, attrsArray, num,  function(err, docs) {
```
Comment (aexmachina) [inline index.js]: Two spaces should be one: `...num, function(err, docs) {`

**Comment by aexmachina** [comment]
Nice one @vichle, just two minor fixes and I'll press the big green button :)

**Comment by vichle** [comment]
Thanks for having me @aexmachina :)

**Comment by aexmachina** [comment]
Nice one, I think this is the point where I need to redo the documentation. But this is fine for now. Thanks!

**Comment by aexmachina** [comment]
Published v1.5.0

**Pull request merged**